### ISSUE TITLE
Flooding TextEditorVisibleRangesChangeEvents

### ISSUE THREAD

**Issue by md2perpe** [Microsoft/vscode#48799]
- VSCode Version: (from Help|About)
Version 1.22.2
Commit 3aeede733d9a3098f7b4bdc1f66b63b0f48c1ef9
Date 2018-04-12T16:34:44.222Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

- OS Version: (from uname -a)
Linux mint 4.10.0-38-generic #42~16.04.1-Ubuntu SMP Tue Oct 10 16:32:20 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
The OS is run from a USB stick.

Update: Also verified on Windows 10 with same VSCode version.

Steps to Reproduce:
See https://github.com/md2perpe/vscode.bug.demo

Does this issue occur when all extensions are disabled?:
Difficult to tell. How can I see if an event is raised without having an extension logging it?
But my demo extension is the only extension loaded.

**Comment by md2perpe** [comment]
## Hypothesis
Is the event sent also when an output channel (e.g. Output > demo) is updated?

**Comment by md2perpe** [comment]
The problem is that I didn't realize that the output channels are considered to be text editors. Therefore I thought that VS Code was stampeding.

Now that I know that also output channels are text editors, it's obvious why I got this behavior.

This was not code meant to be in production. It was just during testing of different functions that this showed up.

Closing issue as it was no bug, but just a lack of knowledge.
